### PR TITLE
[FLINK-5027] FileSource finishes successfully with a wrong path

### PR DESCRIPTION
**Pull request by kl0u**
Adds a test in the `run()` of the `ContinuousFileMonitoringFunction` that checks if the user-specified path is valid. If not, the job will throw an IOException at runtime. We cannot throw the exception at compile time as we may not have access to the cluster.

R: @tillrohrmann

### CODE REVIEW

File: flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java [hunk 1]
```diff
@@ -124,6 +124,9 @@ public void open(Configuration parameters) throws Exception {
 	@Override
 	public void run(SourceFunction.SourceContext<TimestampedFileInputSplit> context) throws Exception {
 		FileSystem fileSystem = FileSystem.get(new URI(path));
+		if (!fileSystem.exists(new Path(path))) {
+			throw new IOException("The provided file path " + path + " does not exist.");
```
Comment (tillrohrmann) [inline flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/source/ContinuousFileMonitoringFunction.java]: I think we should throw an `FileNotFoundException` which better reflects the reason of the error.

**Comment by kl0u** [comment]
@tillrohrmann Thanks for the review. I integrated the comments.

**Comment by tillrohrmann** [comment]
Thanks for your contribution @kl0u. Will merge the PR.

**Comment by kl0u** [comment]
Thanks a lot @tillrohrmann

**Pull request closed without merging**